src/components/GlobalPreferences/Network/Network.js

```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components'
+import { AragonType } from '../../../prop-types'
+import { Box, Button, Info, GU, Text, TextInput, theme } from '@aragon/ui'
+import { InvalidNetworkType, InvalidURI, NoConnection } from '../../../errors'
+import { sanitizeNetworkType } from '../../../network-config'
+import { defaultEthNode, ipfsDefaultConf, network } from '../../../environment'
+import { checkValidEthNode } from '../../../web3-utils'
+import { setDefaultEthNode, setIpfsGateway } from '../../../local-settings'
+
+function Network({ wrapper }) {
+  const {
+    ethNode,
+    ipfsGateway,
+    handleNetworkChange,
+    handleClearCache,
+    networkError,
+    handleEthNodeChange,
+    handleIpfsGatewayChange,
+    network,
+  } = useNetwork(wrapper)
+
+  return (
+    <React.Fragment>
+      <Box heading={'Node settings'}>
+        <Label>
+          Ethereum node{' '}
+          <TextInput
+            value={ethNode}
+            wide
+            onChange={handleEthNodeChange}
+            css={networkError && `border-color: ${theme.negative};`}
+          />
+          {networkError && (
+            <Text color={theme.negative} size="xsmall">
+              {(() => {
+                if (networkError instanceof InvalidNetworkType) {
+                  return `Node must be connected to ${sanitizeNetworkType(
+                    network.type
+                  )}`
+                }
+                if (networkError instanceof InvalidURI) {
+                  return 'Must provide WebSocket endpoint to node'
+                }
+                if (networkError instanceof NoConnection) {
+                  return 'Could not connect to node'
+                }
+                return 'URI does not seem to be a ETH node'
+              })()}
+            </Text>
+          )}
+        </Label>
+        <Label>
+          IPFS Gateway{' '}
+          <TextInput
+            value={ipfsGateway}
+            wide
+            onChange={handleIpfsGatewayChange}
+          />
+        </Label>
+        <Button mode="strong" onClick={handleNetworkChange}>
+          Save changes
+        </Button>
+      </Box>
+      <Box heading={'Troubleshooting'}>
+        <div
+          css={`
+            margin-bottom: ${2 * GU}px;
+          `}
+        >
+          <Text>
+            Press this button to refresh the cache of the application in your
+            browser.
+          </Text>
+        </div>
+        <Button
+          css={`
+            margin-bottom: ${2 * GU}px;
+          `}
+          onClick={handleClearCache}
+        >
+          Clear application cache
+        </Button>
+        <Info>
+          This will only delete the data stored in youur browser to make the app
+          load faster. No data related to the organization itself will be
+          altered.
+        </Info>
+      </Box>
+    </React.Fragment>
+  )
+}
+
+Network.propTypes = {
+  wrapper: AragonType,
+}
+
+const useNetwork = wrapper => {
```

Comment:
Do you think we could move this somewhere higher? It might be useful from other components in the future :smile:

Reply:
For the time being I don't see a need to move it higher, but yeah, let's keep an eye out in case we find that need.